pkg/modulewriter/modulewriter.go

```diff
@@ -112,17 +115,25 @@ func WriteDeployment(blueprint *config.Blueprint, outputDir string, overwriteFla
 	return nil
 }
 
-func copySource(deploymentPath string, deploymentGroups *[]config.DeploymentGroup) {
+func copySource(deploymentPath string, deploymentGroups *[]config.DeploymentGroup) error {
+
 	for iGrp, grp := range *deploymentGroups {
+		basePath := filepath.Join(deploymentPath, grp.Name)
+		// Create the deployment group directory if not already created.
+		if _, err := os.Stat(basePath); errors.Is(err, os.ErrNotExist) {
+			deploymentio := deploymentio.GetDeploymentioLocal()
+			if err := deploymentio.CreateDirectory(basePath); err != nil {
+				return fmt.Errorf("failed to create directory at %s for deployment group %s: err=%w", basePath, grp.Name, err)
```

Comment:
Could you write a unit test for this? If it is too closely tied to copying source then maybe having it as its own function `createGroupDirs()` would make it easier to test.

Reply:
Done